internal/app/machined/pkg/runtime/v1alpha1/platform/metal/metal.go

```diff
@@ -48,26 +49,25 @@ func (m *Metal) Configuration(ctx context.Context) ([]byte, error) {
 
 	u, err := url.Parse(*option)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to parse talos.config url: %w", err)
```

Comment:
We have a constant somewhere for talos.config. We should probably use that just in case we ever need to change it.